sound/soc/sof/ipc4-topology.c

```diff
@@ -1735,8 +1740,9 @@ static int sof_ipc4_prepare_process_module(struct snd_sof_widget *swidget,
 	if (ret < 0)
 		return ret;
 
-	memcpy(&process->output_format, &available_fmt->output_pin_fmts[ret].audio_fmt,
-	       sizeof(struct sof_ipc4_audio_format));
+	if (available_fmt->output_pin_fmts)
+		memcpy(&process->output_format, &available_fmt->output_pin_fmts[ret].audio_fmt,
+		       sizeof(struct sof_ipc4_audio_format));
```

Comment:
> let's say you got a match with the 2nd input format in `sof_ipc4_init_audio_fmt()`, that will make ret==1, let's say you have one output pin. `available_fmt->output_pin_fmts[1]` is out of bound.

Yes, I mentioned a similar question in https://github.com/thesofproject/linux/pull/4163#discussion_r1095409166. The output is not 1:1 mapping to input. Sometimes input number may be larger than output number. This may cause issue. What I understand is we will limit it in tplg based on @ranj063 's comments?

Reply:
As @ujfalusi raise this issue, I still want to recommend that we can add an index in the format structure. Let say:
```
 struct sof_ipc4_pin_format {
        u32 pin_index;
+       u32 format_index;
        u32 buffer_size;
        struct sof_ipc4_audio_format audio_fmt;
 };
```
When there is output format mapped to an input format, they have the same format_index. So we don't use ```ret``` as the output fmt idx. This can help us to avoid such out of bound accessing.